### DISCUSSION TITLE
create own shared version number for multiple dependencies

### DISCUSSION THREAD

**Discussion by nkiesel** [Splitties/refreshVersions#520] [Q&A]
Hello,
we internally generate 30+ jars which all share the same version number:
```koltin
public object Libs {
    public const val ms_prt: String = "com.metricstream.egrcp:ms-prt:_"

    public const val ms_pus: String = "com.metricstream.egrcp:ms-pus:_"

    public const val ms_pweb: String = "com.metricstream.egrcp:ms-pweb:_"
}
```
```
version.com.metricstream.egrcp..ms-prt=2022.03.0.0.0.0.0-SNAPSHOT
version.com.metricstream.egrcp..ms-pus=2022.03.0.0.0.0.0-SNAPSHOT
version.com.metricstream.egrcp..ms-pweb=2022.03.0.0.0.0.0-SNAPSHOT
```
Is there a way to only have one version (e.g. `version.com.metricstream.egrcp=2022.03.0.0.0.0.0-SNAPSHOT`)?

**Comment by nkiesel** [comment]
I tried to add 
```
refreshVersions {
    extraArtifactVersionKeyRules(file("refreshVersions-metricstream-rules.txt"))
}
```
to `settings.gradle` with content
```
com.metricstream.egrcp.*
    ^^^^^^^^^^^^.^^^^^
```
but that does not seem to work: `./gradlew refreshversionsDependenciesMapping` does not show any mappings for my libs.

Perhaps I misunderstand the syntax in the rules files. Is there any documentation of the format?

**Reply by imashnake0** [reply]
Can you try changing
```
com.metricstream.egrcp.*
    ^^^^^^^^^^^^.^^^^^
```
to
```
com.metricstream.egrcp:*
    ^^^^^^^^^^^^.^^^^^
```

**Reply by nkiesel** [reply]
Unfortunately makes no difference.

**Reply by imashnake0** [reply]
Just making sure but you are running the `refreshVersionsMigrate` task after you make your changes yes?

**Reply by nkiesel** [reply]
hah: I did not do that.  After running that, I now have a `version.metricstream.egrcp=2022.03.0.0.0.0.0-SNAPSHOT`! I thought this task would only be required to be run once after plugin upgrade. 

All the existing `version.com.metricstream.egrcp..ms-prt=2022.03.0.0.0.0.0-SNAPSHOT` stayed though. I deleted them manually and that seems to work.

I'm still struggling to understand the "_" resolution process. The ancient buildSrcVersions 0.7.0 version I am trying to upgrade from made it very simple to understand which exact version is used for which dependency. With 0.40.1, looks like I have to understand the content of rules files (both built-in and my own) to guess which version is actually used.  Also, `./gradlew refreshVersionsDependenciesMapping` still does not show my "egrcp" rule. Why is that?
